webapi/src/main/scala/org/knora/webapi/messages/v2/responder/listsmessages/ListsMessagesV2.scala

```diff
@@ -0,0 +1,221 @@
+                                makeNode(childNode) // recursion
+                        })
+                )
+            } else {
+                // no children (abort condition for recursion)
+                Map.empty[IRI, JsonLDArray]
+            }
+
+            JsonLDObject(
+                Map(
+                    "@id" -> JsonLDString(node.id),
+                    "@type" -> JsonLDString(OntologyConstants.KnoraBase.ListNode.toSmartIri.toOntologySchema(ApiV2WithValueObjects).toString)
+                ) ++ position ++ nodeHasRootNode ++ children ++ label ++ comment
+            )
+        }
+
+        val label: Map[IRI, JsonLDString] = makeMapIriToJSONLDString(OntologyConstants.Rdfs.Label, list.listinfo.labels, userLang, fallbackLang)
+
+        val comment: Map[IRI, JsonLDString] = makeMapIriToJSONLDString(OntologyConstants.Rdfs.Comment, list.listinfo.comments, userLang, fallbackLang)
+
+        val children: Map[IRI, JsonLDArray] = if (list.children.nonEmpty) {
+            Map(
+                OntologyConstants.KnoraBase.HasSubListNode.toSmartIri.toOntologySchema(ApiV2WithValueObjects).toString -> JsonLDArray(list.children.map {
+                    childNode: ListNodeADM =>
+                        makeNode(childNode)
+                }))
+        } else {
+            Map.empty[IRI, JsonLDArray]
+        }
+
+        val project: Map[IRI, JsonLDObject] = Map(
+            OntologyConstants.KnoraBase.AttachedToProject.toSmartIri.toOntologySchema(ApiV2WithValueObjects).toString -> JsonLDUtil.iriToJsonLDObject(list.listinfo.projectIri)
+        )
+
+        val body = JsonLDObject(Map(
+            "@id" -> JsonLDString(list.listinfo.id),
+            "@type" -> JsonLDString(OntologyConstants.KnoraBase.ListNode.toSmartIri.toOntologySchema(ApiV2WithValueObjects).toString),
+            OntologyConstants.KnoraBase.IsRootNode.toSmartIri.toOntologySchema(ApiV2WithValueObjects).toString -> JsonLDBoolean(true)
+        ) ++ project ++ children ++ label ++ comment)
+
+        val context = JsonLDObject(Map(
+            OntologyConstants.KnoraApi.KnoraApiOntologyLabel -> JsonLDString(OntologyConstants.KnoraApiV2WithValueObjects.KnoraApiV2PrefixExpansion),
+            "rdfs" -> JsonLDString("http://www.w3.org/2000/01/rdf-schema#"),
+            "rdf" -> JsonLDString("http://www.w3.org/1999/02/22-rdf-syntax-ns#"),
+            "owl" -> JsonLDString("http://www.w3.org/2002/07/owl#"),
+            "xsd" -> JsonLDString("http://www.w3.org/2001/XMLSchema#")
+        ))
+
+        JsonLDDocument(body, context)
+    }
+}
+
+/**
+  * Requests a list node. A successful response will be a [[NodeGetResponseV2]]
+  *
+  * @param nodeIri the IRI of the node to retrieve.
+  */
+case class NodeGetRequestV2(nodeIri: IRI,
+                            requestingUser: UserADM) extends ListsResponderRequestV2
+
+/**
+  * Represents a response to a [[NodeGetRequestV2]].
+  *
+  * @param node the node to be returned.
```

Comment:
Missing parameter descriptions.

Reply:
done in https://github.com/dhlab-basel/Knora/pull/852/commits/8bba915dd4a23512b68d5dde430551ac7de5c82c